src/views/1_ApplicationComponent.tsx

```diff
@@ -9,33 +9,77 @@ import {saveWindowBounds} from "./ViewUtils";
 import {StatusBarComponent} from "./StatusBarComponent";
 import {PaneTree, Pane} from "../utils/PaneTree";
 import {SearchComponent} from "./SearchComponent";
+import {SplitDirection} from "../Enums";
 
 export class ApplicationComponent extends React.Component<{}, {}> {
     private tabs: Tab[] = [];
     private focusedTabIndex: number;
+    private electronWindow: Electron.BrowserWindow = remote.getCurrentWindow();
 
     constructor(props: {}) {
         super(props);
-        const electronWindow = remote.BrowserWindow.getAllWindows()[0];
 
         this.addTab(false);
 
-        electronWindow
-            .on("move", () => saveWindowBounds(electronWindow))
+        this.electronWindow
+            .on("move", () => saveWindowBounds(this.electronWindow))
             .on("resize", () => {
-                saveWindowBounds(electronWindow);
+                saveWindowBounds(this.electronWindow);
                 this.recalculateDimensions();
-            })
-            .webContents
+            });
+
+        this.electronWindow.webContents
             .on("devtools-opened", () => this.recalculateDimensions())
             .on("devtools-closed", () => this.recalculateDimensions());
 
-        ipcRenderer.on("change-working-directory", (_event: Electron.IpcRendererEvent, directory: string) =>
-            this.focusedTab.focusedPane.session.directory = directory,
-        );
+        ipcRenderer
+            // Tabs actions
+            .on("change-working-directory", (_event: Electron.IpcRendererEvent, directory: string) =>
```

Comment:
Ideally I'd like to have direct function calls instead of events, but I'm not sure it's possible. An enum would be nice too. Alternatively you can extend the interface of the `on` function to accept particular strings. This way you can verify the second argument based on the first one.